src/model/APNSRequestMetadata.php

```diff
@@ -179,18 +179,40 @@ public function toJSON(): string {
 	public static function fromJSON( string $data ): self {
 		$object = json_decode( $data );
 
-		$topic = $object->topic;
-
-		if ( is_null( $topic ) ) {
+		if ( ! property_exists( $object, 'topic' ) || is_null( $object->topic ) ) {
 			throw new InvalidArgumentException( 'Unable to unserialize object – `topic` is not present' );
 		}
 
-		$uuid = $object->uuid;
-
-		if ( is_null( $uuid ) ) {
+		if ( ! property_exists( $object, 'uuid' ) || is_null( $object->uuid ) ) {
 			throw new InvalidArgumentException( 'Unable to unserialize object – `uuid` is not present' );
 		}
 
-		return new APNSRequestMetadata( $topic, $uuid );
+		$metadata = new APNSRequestMetadata( $object->topic, $object->uuid );
+
+		if ( property_exists( $object, 'push_type' ) && ! is_null( $object->push_type ) ) {
+			$metadata->setPushType( $object->push_type );
+		}
+
+		if ( property_exists( $object, 'expiration_timestamp' ) && ! is_null( $object->expiration_timestamp ) ) {
+			$metadata->setExpirationTimestamp( $object->expiration_timestamp );
+		}
+
+		if ( property_exists( $object, 'priority' ) && ! is_null( $object->priority ) ) {
+			if ( ! APNSPriority::isValid( $object->priority ) ) {
+				throw new InvalidArgumentException( 'Unable to unserialize object – `priority` is invalid' );
+			}
+
+			if ( $object->priority === APNSPriority::IMMEDIATE ) {
+				$metadata->setNormalPriority();
+			} elseif ( $object->priority === APNSPriority::THROTTLED ) {
+				$metadata->setLowPriority();
+			}
```

Comment:
Would it be worth adding a final `elseif` that throws, or is it enough to have the `isValid` check above?

Another way to look at it, if we get a new `APNSPriority` value, how many places in the codebase do we need to change?

Anyways, this is just a nitpick and I don't have an actual answer for this...

Reply:
Yeah, that's a good question – we're sort of belt-and-suspenders-ing this particular check. I don't mind that, but it might be cleaner. I don't expect Apple to add more values any time soon. 

I think the main check is that we'd see code coverage go down and want to know why. 

But I'd hope someone would write a failing test case first ;)

Reply:
> I don't expect Apple to add more values any time soon.
True
> I think the main check is that we'd see code coverage go down and want to know why.
>
> But I'd hope someone would write a failing test case first ;)
Also true 😄